HW3.4.java

```diff
@@ -1 +1,43 @@
+package com.company;
 
+public final class ComplexNumberApp {
+
+    private final double re;
+    private final double im;
+
+    public static void main(String[] args) {
+        ComplexNumberApp firstNumber = new ComplexNumberApp(1, 1);
+        ComplexNumberApp secondNumber = new ComplexNumberApp(1, 1);
+        System.out.print("firstNumber.equals(secondNumber) is ");
+        System.out.println(firstNumber.equals(secondNumber));
+        System.out.print("firstNumber.hashCode() ==  secondNumber.hashCode() is ");
+        System.out.println(firstNumber.hashCode(firstNumber.getRe(), firstNumber.getIm()) == secondNumber.hashCode(secondNumber.getRe(),secondNumber.getIm()));
+    }
+
+    public boolean equals(ComplexNumber complexNumber) {
+        return (this.getIm() == complexNumber.getIm()) && (this.getRe() == complexNumber.getRe());
```

Comment:
it is not full realization
read about equals and hashCode
https://javarush.ru/groups/posts/1340-peregruzka-metodov-equals-i-hashcode-v-java

Reply:
ok, thank you